lambdas/tests/unit/services/token-request-validator.test.ts

```diff
@@ -0,0 +1,168 @@
+import { AccessTokenRequestValidator } from "../../../src/services/token-request-validator";
+import { JwtVerifierFactory } from "../../../src/common/security/jwt-verifier";
+import { SessionItem } from "../../../src/types/session-item";
+describe("token-request-validator.ts", () => {
+    let accessTokenRequestValidator: AccessTokenRequestValidator;
+    const mockJwtVerifierFactory = jest.mocked(JwtVerifierFactory);
+    const code = "test";
+    const redirect_uri = "http://abc123.com";
+    const client_assertion = "test";
+    const client_assertion_type = "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
+    const grant_type = "authorization_code";
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+        accessTokenRequestValidator = new AccessTokenRequestValidator(mockJwtVerifierFactory.prototype);
+    });
+
+    describe("validatePayload", () => {
+        it("should throw when the client_assertion_type is not valid", function () {
```

Comment:
Did you check that the grant_type and client_assertion_type was a mismatch in the java code too like the guys in slack recommended?

Reply:
Was planning on leaving this as is for now since if it is a bug, it will need to be fixed in another ticket